### PR TITLE
Add right click->context menu->copy functionality to event logs

### LINKED ISSUE #11100: Allow to right-click to copy log content from Help → View Event log
**Is your suggestion for improvement related to a problem? Please describe.**
When I want to report an issue and I would like to copy the log content, I cannot copy it via right click.

**Describe the solution you'd like**
I would like to copy the log entry by right click -> Copy

**Additional context**
log

### PR DESCRIPTION
**Pull request by subhramit**
This closes #11100 
Users can now right click on a log entry from the event log window to copy the log entry using a context menu:

Screenshot (220)
(Note: Windows screenshots/print screen do not show the mouse cursor location)
### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- ~[ ] Tests created for changes (if applicable)~
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java [hunk 1]
```diff
@@ -112,6 +114,17 @@ public void updateItem(LogEventViewModel event, boolean empty) {
                     setGraphic(graphic);
                 }
             }
+
+            {
```
Comment (koppor) [inline src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java:118]: Why this opening and closing brace below? Should that block go to a separate method?

Comment (subhramit) [reply src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java:118]: I was defining an instance initializer block for an anonymous inner class that is being created as the cell factory for the ```messagesListView```.
Good point though, that makes it very badly readable, I'll move it to a separate method instead

File: src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java [hunk 2]
```diff
@@ -102,6 +104,7 @@ public void updateItem(LogEventViewModel event, boolean empty) {
 
                 if ((event == null) || empty) {
                     setGraphic(null);
+                    setContextMenu(null);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java:107]: Why this call? What is the behavior if this is removed? I would not show any context menu if something is empty.

Then, in src/main/java/org/jabref/gui/errorconsole/ErrorConsoleViewModel.java, you can remove the `null` check.

File: src/main/java/org/jabref/gui/errorconsole/ErrorConsoleViewModel.java [hunk 3]
```diff
@@ -78,6 +78,16 @@ public void copyLog(List<LogEventViewModel> messages) {
         dialogService.notify(Localization.lang("Log copied to clipboard."));
     }
 
+    /**
+     * Copies the detailed text of the given {@link LogEventViewModel} to the clipboard.
+     */
+    public void copyLogEntry(LogEventViewModel logEvent) {
+        if (logEvent != null) {
+            clipBoardManager.setContent(logEvent.getDetailedText());
+            dialogService.notify(Localization.lang("Log entry copied to clipboard."));
```
Comment (koppor) [inline src/main/java/org/jabref/gui/errorconsole/ErrorConsoleViewModel.java:87]: Please remove this.

I know that JabRef has notifications for copy operations. I personally find this strange. Moreover, the log scrolls down after copying. Thus, in this case, it is annoying.

I think, we won't have issues when copying to clipboard. Thus, no notification needed.

**Review by koppor — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java (hunk 2 above)
Comment (subhramit) [reply src/main/java/org/jabref/gui/errorconsole/ErrorConsoleView.java]: Makes sense. Removed.

File: src/main/java/org/jabref/gui/errorconsole/ErrorConsoleViewModel.java (hunk 3 above)
Comment (subhramit) [reply src/main/java/org/jabref/gui/errorconsole/ErrorConsoleViewModel.java]: Done.

**Review by calixtus — DISMISSED** [review]

**Review by koppor — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**